### ISSUE TITLE
state.artifactory replaces dots in group_id with slashes, results in incorrect URI

### ISSUE THREAD

**Issue by nickgarber** [saltstack/salt#48657]
### Description of Issue/Question

When trying to use the `artifactory.downloaded` state to retrieve a file that resides on our Artifactory at `https://somerepo/artifactory/logstuff-misc-prd-local/co.nxlog/nxlog-community-edition/nxlog-community-edition-2.10.2102.msi` I provide the group_id of `co.nxlog` but the state fails with an error indicating that the group_id value is being split on dots ('.') 

```
local:
----------
          ID: Get NXLog Install Media
    Function: artifactory.downloaded
      Result: False
     Comment: HTTP Error 404. Request URL: https://somerepo/artifactory/logstuff-misc-prd-local/co/nxlog/nxlog-community-edition/2.10.2102/nxlog-community-edition-2.10.2102.msi
     Started: 14:22:29.235815
    Duration: 29.75 ms
     Changes:   
----------
```

I followed the call through these sequential descending steps:
 - state.artifactory.py:downloaded()
 - state.artifactory.py:__fetch_from_artifactory()
 - module.artifactory.py:get_release()
 - module.artifactory.py:_get_release_url()

It's in the `_get_release_url` function that a call to `__get_group_id_subpath(group_id)` replaces the dots with slashes. 

This seems like an assumption about the repo layout that just doesn't hold true in my case. This repo is using the 'simple-default' layout.

I'd submit a PR but not sure what the best approach is. Any thoughts/suggestions?

(I'm using Artifactory Pro version 5.9 and Salt 2018.3.1)

**Comment by nickgarber** [comment]
Thoughts so far ...
It could be that the best way forward is (1) to add and support a `layout` argument, or (2) to allow mashing both together in `repository` and make `group_id` optional.

The second option doesn't currently work because it produces URIs like this `https://somerepo/artifactory/logstuff-misc-prd-local/co.nxlog/u''/nxlog-community-edition/2.10.2102/nxlog-community-edition-2.10.2102.msi`

I'll look more into option 2 for now.

**Comment by nickgarber** [comment]
Oh! Looks like this is already fixed in 2018.3.2 (https://github.com/saltstack/salt/commit/3557c0ddea0be9a37c48a280d26fbb0e65ba47ab)

Cheers and beers!
